src/fs/discover.rs

```diff
@@ -132,16 +132,22 @@ pub fn perform_discovery() -> LaunchPad<StandardLoader> {
             })
             .collect();
 
-        let new_mods: HashSet<&Hash40> = new_cache.iter().filter(|cached_mod| !mod_cache.contains(cached_mod)).collect();
+        // Get the workspace name and workspace list
+        let workspace_name: String = storage.get_field("workspace").unwrap_or("Default".to_string());
+        let workspace_list: HashMap<String, String> = storage.get_field_json("workspace_list").unwrap_or_default();
+
+        // Get the preset name from the 
+        let preset_name = &workspace_list[&workspace_name];
+        let mut presets: HashSet<Hash40> = storage.get_field_json(preset_name).unwrap_or_default();
+        let new_mods: HashSet<&Hash40> = new_cache.iter().filter(|cached_mod| !mod_cache.contains(cached_mod)).filter(|cached_mod| !presets.contains(cached_mod)).collect();
```

Comment:
Could be done in a single filter by using a AND binary operator (&&).
In this case ``!mod_cache.contains(cache_mod) && !presets.contains(cache_mods)`` would only keep the hash if it's not in the cache AND not in the preset.

Reply:
oh sick, tried to find an and operator online but couldn't find one anywhere